api/middlewares.go

```diff
@@ -0,0 +1,83 @@
+// everest
+// Copyright (C) 2023 Percona LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"net/http"
+	"slices"
+	"strings"
+
+	"github.com/AlekSi/pointer"
+	"github.com/labstack/echo/v4"
+
+	everestv1alpha1 "github.com/percona/everest-operator/api/v1alpha1"
+)
+
+func (e *EverestServer) shouldAllowRequestDuringEngineUpgrade(c echo.Context) (bool, error) {
+	// We allow read-only requests.
+	if c.Request().Method == http.MethodGet {
+		return true, nil
+	}
+
+	// List of resources that should not be modified during an upgrade.
+	targetResources := []string{
+		"database-clusters",
+		"database-engines",
+		"database-cluster-restores",
+		"database-cluster-backups",
+	}
```

Comment:
should Everest allow to 
put '/namespaces/{namespace}/database-engines/{name}'
put '/namespaces/{namespace}/database-engines/{name}/operator-version' ?

Reply:
oh, there we don't allow it already. resolving